points/module.py

```diff
@@ -282,6 +166,55 @@ def _get_board(
             result.append(template.format(points=db_user.points, name=name))
         return "\n".join(result)
 
+    @staticmethod
+    def _create_embeds(
+        ctx,
+        title: str,
+        description: str,
+        order: BoardOrder,
+        element_count: int,
+        page_count: int,
+    ) -> list[discord.Embed]:
+        elements = []
+
+        for page_number in range(page_count):
+            users = UserStats.get_best(
+                ctx.guild.id, order, element_count, page_number * element_count
+            )
+
+            if not users:
+                break
+
+            page = utils.Discord.create_embed(
+                author=ctx.author,
+                title=title,
+                description=description,
+            )
+
+            value = Points._get_page(ctx.guild, ctx.author, users)
+
+            page.add_field(
+                name=_(ctx, "Top {limit}").format(limit=(page_count * element_count)),
+                value=value,
+                inline=False,
+            )
+
+            if ctx.author.id not in [u.user_id for u in users]:
+                author = UserStats.get_stats(ctx.guild.id, ctx.author.id)
```

Comment:
We are querying the author for every page, which is not optimal. This line can be moved up before the `for` loop, so it is only looked up once.